site/app/views/forum/ForumThreadView.php

```diff
@@ -841,10 +858,15 @@ public function displayThreadList($threads, $filtering, &$activeThreadAnnounceme
                 $email = trim($user_info['user_email']);
                 $first_name = trim($user_info["first_name"]);
                 $last_name = trim($user_info["last_name"]);
+                $visible_username = $first_name . " " . substr($last_name, 0, 1) . ".";
+
+                if ($this->core->getQueries()->isInstructorOrFullAccess($first_post["author_user_id"])) {
+                    $visible_username = $first_name . " " . $last_name;
+                }
```

Comment:
This query is nested inside a `foreach` loop, which will result in a lot of repetitive queries.  It would be better to make one query which gets all of the necessary data and caches it instead.